src/impl/tpls/KokkosBlas2_gemv_tpl_spec_decl.hpp

```diff
@@ -44,9 +44,456 @@
 #ifndef KOKKOSBLAS2_GEMV_TPL_SPEC_DECL_HPP_
 #define KOKKOSBLAS2_GEMV_TPL_SPEC_DECL_HPP_
 
+#ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS
```

Comment:
BLAS libraries may use different underscore name mangling conventions, so you shouldn't hard-code the underscore.  See Teuchos for an example of how to handle this.

I am not a fan of all these `extern` declarations being exposed to header files that applications will include.  That will almost certainly collide with applications' `extern` declarations.  A better approach would be to wrap BLAS functions in C-style wrappers, and only expose the wrappers.  This would also help hide cuBLAS, MKL, and other TPL header file includes from kokkos-kernels' header files.

Reply:
Thanks @mhoemmen for your comments. This PR meant for supporting CUBLAS so I just followed the way to call BLAS libraries in other functions. I will fix this in another PR.